pkg/state/state_whereabouts.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2020 NVIDIA
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package state
+
+import (
+	"github.com/pkg/errors"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	mellanoxv1alpha1 "github.com/Mellanox/network-operator/pkg/apis/mellanox/v1alpha1"
+	"github.com/Mellanox/network-operator/pkg/consts"
+	"github.com/Mellanox/network-operator/pkg/render"
+	"github.com/Mellanox/network-operator/pkg/utils"
+)
+
+// NewStateWhereabouts creates a new state for Whereabouts
+func NewStateWhereabouts(k8sAPIClient client.Client, scheme *runtime.Scheme, manifestDir string) (State, error) {
+	files, err := utils.GetFilesWithSuffix(manifestDir, render.ManifestFileSuffix...)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get files from manifest dir")
+	}
+
+	renderer := render.NewRenderer(files)
+	return &stateWhereabouts{
+		stateSkel: stateSkel{
+			name:        "state-whereabouts",
+			description: "Deploy whereabouts IpamPlugin for secondary network",
+			client:      k8sAPIClient,
+			scheme:      scheme,
+			renderer:    renderer,
+		}}, nil
+}
+
+type stateWhereabouts struct {
```

Comment:
we got 3 states the deploy CNIs (multus, macvlan, whereabouts) id like to have consistency in the naming scheme of the 3.

so either:
stateMultus
stateWhereabouts
stateMacvlan

or:
stateMultusCNI
stateWhereaboutsCNI
stateMacvlanCNI

Reply:
Changed to stateWhereaboutsCNI